### ISSUE TITLE
ureadahead is still shown as a required package in Mint v20

### ISSUE THREAD

**Issue by endlisnis** [linuxmint/linuxmint#286]
I hope this is the right place to report this specific problem:

After performing an upgrade from v19.3 to v20, I decided to clean up my foreign packages using the "Software Sources" window and "Remove Foreign packages". I've been able to remove a bunch, but some of them are listed as dependencies of essential packages. I find this strange, because I have a separate (fresh) install of v20 and it does not list anything under foreign packages.

In this particular example, "libnih1" is listed as a foreign package, and when I try to remove it, it says that "ureadahead=0.100.0-21" will have to be removed too, and if I let it try that, it tells me that it's an essential package.

Given that I have a fully functional v20 install that does not complain about this, I'm lead to believe one of these things:
(1) "ureadahead" is not really essential.
(2) "ureadahead" does not really require "libnih1".
(3) "libnih1" is not really a foreign package.
(4) Some part of my local package database is wrong/corrupt.
OR
(5) I misunderstand something about how this all works.

I person on the forums suggested that I report it here.

My current system is functional, but I'd like to understand/fix whatever is going on here.

```
System:
  Kernel: 5.4.0-40-generic x86_64 bits: 64 compiler: gcc v: 9.3.0 
  Desktop: Cinnamon 4.6.6 wm: muffin 4.6.2 dm: LightDM 1.30.0 
  Distro: Linux Mint 20 Ulyana base: Ubuntu 20.04 focal 

```

**Comment by k4ffee** [comment]
Seems removing it produces error messages and a short pause while booting 
https://bugs.launchpad.net/ubuntu/+source/ureadahead/+bug/825144 (quite old)

What ureadahead is for:
http://ubuntuguide.net/howto-fix-ureadahead-problem-after-upgrading-to-ubuntu-10-04

**Comment by endlisnis** [comment]
> What ureadahead is for:
> http://ubuntuguide.net/howto-fix-ureadahead-problem-after-upgrading-to-ubuntu-10-04
Thanks for the info.  I read that, and it's interesting, but my understanding is that it's not even part of ubuntu v20, so it couldn't be that important anymore.

**Comment by santa-klaus** [comment]
I also have `ureadahead` listed as a foreign package after upgrading from 19 to 19.3 to 20. 
I don't understand why it is not in the sources for LM20 if it is essential? Is `ureadahead` part of a fresh LM20 install?

**Comment by endlisnis** [comment]
I did a fresh install of LM20 and ureadahead was not part of the installation.  This just looks like a vestige of days gone by which should no longer be listed as a required package.

**Comment by santa-klaus** [comment]
Ah, okay, sorry, I probably misunderstood your prior posts. Have you tried removing it via the command line? I am a bit afraid of running into the same problem as @k4ffee linked...

**Comment by endlisnis** [comment]
I have not tried.  I'm sure it would work, but I'd rather get the Mint team to actually fix the bug so that other people wouldn't run into this.

**Comment by IndigoJay** [comment]
I am following the steps to upgrade from 19.3 to 20.0. Within Software Sources, Maintenance, Remove Foreign Packages. successfully removed the majority of the foreign packages; and am left with:

- initscripts
- insserv
- libcryptsetup4
- libprocps4
- mountall
- sysv-rc

Several comments on this page with similar lists (search 'Foreign'):
https://blog.linuxmint.com/?p=3946

https://forums.linuxmint.com/viewtopic.php?t=325726

**Comment by RodBarnes** [comment]
I just upgraded from 19.3 to 20.  I was able to remove everything except for the following.  Interesting that it indicates these are "installed on your computer but not present in the repositories".  So, why can't they be removed if 20 no longer needs them?

- libnih1
- libprocps6
- ureadahead

As with the OP, I can't remove libnih1 because it is required by ureadahead and attempts to remove ureadahead "Failed to remove essential package -- Package ureadahead cannot be removed"

I get the same message when attempting to remove libprocps6.

**Comment by endlisnis** [comment]
After several months without any comments by the developers, I gave up waiting and did a fresh re-install of v20.1 from scratch.

I guess that having a fully functional upgrade system is not that important to the linux mint team.

**Comment by endlisnis** [comment]
It was not these 3 packages that caused me to decide to do a reinstall.  I ran into several other problems: One of my printers stopped working, and my computer would fail to reboot (it would get stuck trying to disable some driver).

**Comment by AntarSidgi** [comment]
sudo dpkg --purge --force-all package_name

**Comment by endlisnis** [comment]
> sudo dpkg --purge --force-all package_name

Yes, I'm sure there are work-arounds for this problem, but this bug is really trying to cover the upgrade process.  I describes steps to get rid of older packages, and those steps don't work.  Either the upgrade process needs to be changed, or the system has to be modified such that the existing steps work.
